### ISSUE TITLE
@at-root registers as nested depth

### ISSUE THREAD

**Issue by gitnik** [sasstools/sass-lint#383]
```scss
.foo {
    @at-root &__bar {
        @at-root &__baz {
            display: inline-block;
        }
    }
}
```

This registers as a depth of 2 which is technically correct but considering that `@at-root` generates a completely new, unnested selector I think this rule should be disabled in this specific case. I am willing to submit a PR if this behaviour is not wanted anymore.

**Comment by DanPurdy** [comment]
Hi @gitnik I think this makes sense yes, my only worry is that it opens a lot of edge cases around the rule that we'll need to be very careful with. @sasstools/sass-lint-contributors does anyone have any other opinions on this regarding whether we are really linting compiled nest depth or actual nesting depth within the scss/sass files where I would still class the above as nesting depth of 3.

**Comment by gitnik** [comment]
Well I don't really consider this "nesting" since the end result will be everything but nested. But I can see where you're coming from. Looking forward to what other people have to say about this

**Comment by DanPurdy** [comment]
I had another think about this @gitnik I think you're right it should be excluded. Happy for you to get a PR up if you want to.

**Comment by bgriffith** [comment]
Are we saying that the following should be valid? 

```scss
.foo {
@at-root &__bar {
@at-root &__baz {
    display: inline-block;
}
}
}
```

**Comment by gitnik** [comment]
IMO, yes

**Comment by benthemonkey** [comment]
The use of `@at-root` is extraneous in this example. Suffix extensions are not nested within their parents (so removing `@at-root` outputs the same CSS, [example gist](https://gist.github.com/benthemonkey/b089848bce1833381598)). So if we are to be consistent, we need to treat the use of `@at-root` and suffix extensions identically.

**Comment by brunowego** [comment]
Take a look into my case:

```bash
./src/styles/_form.scss
  46:1  error  Please check validity of the block starting from line #46  Fatal
```

```scss
.form-input {

    @at-root div#{&} {

    }

    @-moz-document url-prefix() {
        @at-root div#{&} {

        }
    }

    @at-root textarea#{&} {

    }
}
```

**Comment by DanPurdy** [comment]
@brunowego this again is the AST you need to report this there

https://github.com/tonyganch/gonzales-pe/issues

We don't have any control over these parse issues unfortunately

**Comment by brunowego** [comment]
Thanks @DanPurdy :+1:
